### PR TITLE
Improvements on SSL generation, adding SAN extension and file permissions

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Adds SAN ext to certs, also sets file permissions based on the service

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested across our 3 ssl options- self signed, custom certs and custom keystores

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: roles/confluent.kafka_broker/tasks/main.yml [hunk 1]
```diff
@@ -1,38 +1,39 @@
...
-    group: "{{kafka_broker.group}}"
-
-- import_role:
+# - name: Install the Kafka Broker Packages
+#   yum:
+#     name: "{{item}}{{confluent_package_redhat_suffix}}"
+#     state: latest
+#   loop: "{{kafka_broker_packages}}"
+#   when: ansible_os_family == "RedHat"
+#
```
Comment (domenicbove) [inline roles/confluent.kafka_broker/tasks/main.yml]: fix this!

File: roles/confluent.ssl-old/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+keystore_expiration_days: 365
```
Comment (domenicbove) [inline roles/confluent.ssl-old/defaults/main.yml]: remove this role

**Comment by domenicbove** [comment]
@JumaX finished testing, and updated that section

File: roles/confluent.ssl/tasks/certificate_authority.yml [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
+# - name: Create CA
+#   shell: |
+#     openssl req -new -x509 \
+#       -keyout /var/ssl/private/generation/{{ssl_self_signed_ca_key_filename}} \
+#       -out /var/ssl/private/generation/{{ssl_self_signed_ca_cert_filename}} \
+#       -days 365 \
+#       -subj '/CN=ca1.test.confluent.io/OU=TEST/O=CONFLUENT/L=PaloAlto/S=Ca/C=US' \
+#       -passin pass:{{ssl_self_signed_ca_password}} -passout pass:{{ssl_self_signed_ca_password}}
+#
+# - name: Create Truststore and Import the CA Cert
```
Comment (JumaX) [inline roles/confluent.ssl/tasks/certificate_authority.yml]: @domenicbove Should this not be removed if we have moved it into another file now?

**Pull request merged**